internal/k8s/appprotectdos/app_protect_dos_configuration.go

```diff
@@ -0,0 +1,271 @@
+package appprotectdos
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// reasons for invalidity
+const (
+	failedValidationErrorMsg = "Validation Failed"
+)
+
+var (
+	// DosPolicyGVR is the group version resource of the appprotectdos policy
+	DosPolicyGVR = schema.GroupVersionResource{
+		Group:    "appprotectdos.f5.com",
+		Version:  "v1beta1",
+		Resource: "apdospolicies",
+	}
+
+	// DosPolicyGVK is the group version kind of the appprotectdos policy
+	DosPolicyGVK = schema.GroupVersionKind{
+		Group:   "appprotectdos.f5.com",
+		Version: "v1beta1",
+		Kind:    "APDosPolicy",
+	}
+
+	// DosLogConfGVR is the group version resource of the appprotectdos policy
+	DosLogConfGVR = schema.GroupVersionResource{
+		Group:    "appprotectdos.f5.com",
+		Version:  "v1beta1",
+		Resource: "apdoslogconfs",
+	}
+	// DosLogConfGVK is the group version kind of the appprotectdos policy
+	DosLogConfGVK = schema.GroupVersionKind{
+		Group:   "appprotectdos.f5.com",
+		Version: "v1beta1",
+		Kind:    "APDosLogConf",
+	}
+)
+
+// Operation defines an operation to perform for an App Protect Dos resource.
+type Operation int
+
+const (
+	// Delete the config of the resource
+	Delete Operation = iota
+	// AddOrUpdate the config of the resource
+	AddOrUpdate
+)
+
+// Change represents a change in an App Protect Dos resource
+type Change struct {
+	// Op is an operation that needs be performed on the resource.
+	Op Operation
+	// Resource is the target resource.
+	Resource interface{}
+}
+
+// Problem represents a problem with an App Protect Dos resource
+type Problem struct {
+	// Object is a configuration object.
+	Object *unstructured.Unstructured
+	// Reason tells the reason. It matches the reason in the events of our configuration objects.
+	Reason string
+	// Message gives the details about the problem. It matches the message in the events of our configuration objects.
+	Message string
+}
+
+// Configuration configures App Protect Dos resources that the Ingress Controller uses.
+type Configuration interface {
+	AddOrUpdateDosPolicy(policyObj *unstructured.Unstructured) (changes []Change, problems []Problem)
```

Comment:
Because the dos part is already in the name of the package, I suggest removing the Dos part from the names of all methods of this interface.

Reply:
Done